cvmfs/catalog_rw.cc

```diff
@@ -647,6 +647,26 @@ void WritableCatalog::MergeIntoParent() {
 }
 
 
+void WritableCatalog::RemoveFromParent() {
+  assert(!IsRoot() && HasParent());
+  WritableCatalog *parent = GetWritableParent();
+
+  // Remove the nested catalog reference for this nested catalog.
+  // From now on this catalog will be dangling!
+  Catalog* child_catalog;
+  parent->RemoveNestedCatalog(this->mountpoint().ToString(), &child_catalog);
+
+  const Counters& child_counters = child_catalog->GetCounters();
+
+  parent->delta_counters_.subtree.directories = -1;
```

Comment:
The `parent->delta_counters_` could already have some non-zero value from previous operations which would get overwritten here, no?

Reply:
Most likely. I've been trying to understand how these counters work, by reading the code.